test/evp_kdf_test.c

```diff
@@ -1137,6 +1141,30 @@ static int test_kdf_kbkdf_invalid_mac(void)
     return ret;
 }
 
+static int test_kdf_kbkdf_invalid_r(void)
```

Comment:
I think the unit tests would be better in `evp_test` with data in a file in `test/recipes/30-test_evp_data`.  It doesn't look like we've got any KBKDF test cases there & there really ought to be some.  Data driven tests are almost always better than ad-hoc ones.

Reply:
That is much more usable yes, thank you very much for pointing that test out!
(I had been looking at `test/recipes/20-test_kdf.t` prior, and that didn't look really nice either, but `evp_test` is perfect!)